imagej/imagej-ops2/src/main/java/net/imagej/ops2/types/ImgFactoryTypeExtractor.java

```diff
@@ -48,25 +49,21 @@
  *
  * @author Gabriel Selzer
  */
-public class ImgFactoryTypeExtractor implements TypeExtractor<ImgFactory<?>> {
+public class ImgFactoryTypeExtractor implements TypeExtractor {
 
-	@Override
-	public Type reify(final TypeReifier t, final ImgFactory<?> o, final int n) {
-		if (n != 0)
-			throw new IndexOutOfBoundsException();
-
-		return t.reify(o.type());
+	@Override public double getPriority() {
+		return Priority.NORMAL;
 	}
 
-	@Override
-	@SuppressWarnings({ "rawtypes", "unchecked" })
-	public Class<ImgFactory<?>> getRawType() {
-		return (Class) ImgFactory.class;
+	@Override public boolean canReify(TypeReifier r, Class<?> object) {
+		return ImgFactory.class.isAssignableFrom(object);
 	}
 
-	@Override
-	public double priority() {
-		return Priority.LOW;
+	@Override public Type reify(TypeReifier r, Object object) {
+		if (!(object instanceof ImgFactory))
+			throw new IllegalArgumentException(this + " cannot reify " + object);
+		ImgFactory<?> imgFactory = (ImgFactory<?>) object;
+		Type componentType = r.reify(imgFactory.type());
+		return TypeTools.raiseParametersToClass(object.getClass(), ImgFactory.class, new Type[] {componentType});
```

Comment:
In general, I love the generalization of the `reify` method. But I dislike this boilerplate `TypeTools.raiseParametersToClass` thing that is now needed in the majority of `TypeExtractor` implementations. Surely we can make the code more DRY somehow?

Reply:
Made an abstract superclass in 4e897c7c